### PR TITLE
add doc about CANN EP

### PR DESCRIPTION
**Pull request by FFFrog**
#### Description
This PR adds docs of Ascend CANN excution provider.

#### Changes

- Preview Github page: [https://fffrog.github.io/](https://fffrog.github.io/)

- Add onnxruntime build with CANN: [https://fffrog.github.io/docs/build/eps.html#cann](https://fffrog.github.io/docs/build/eps.html#cann)

- Add CANN ExecutionProvider Page: [https://fffrog.github.io/docs/execution-providers/CANN-ExecutionProvider.html](https://fffrog.github.io/docs/execution-providers/CANN-ExecutionProvider.html)

### CODE REVIEW

**Comment by FFFrog** [comment]
@microsoft-github-policy-service agree company="Huawei"

File: docs/build/eps.md [hunk 1]
```diff
@@ -768,4 +768,37 @@ Linux example:
...
+
+## CANN
+
+See more information on the CANN Execution Provider [here](../execution-providers/CANN-ExecutionProvider.md).
+
+### Prerequisites
+{: .no_toc }
+
+1. Install the CANN Toolkit for the appropriate OS and target hardware by following [documentation](https://www.hiascend.com/document/detail/zh/CANNCommunityEdition/60RC1alpha003/softwareinstall/instg/atlasdeploy_03_0018.html) for detailed instructions, please.
+
```
Comment (wangxiyuan) [inline docs/build/eps.md:783]: please use english version for the link

Comment (FFFrog) [reply docs/build/eps.md:783]: Done

**Review by wangxiyuan — APPROVED** [review]

**Comment by FFFrog** [comment]
Sorry to @jywu-msft again, the commit of docs on CANN EP is ready. Could you please take a look at this PR? Thanks a lot.

**Comment by jywu-msft** [comment]
> Sorry to @jywu-msft again, the commit of docs on CANN EP is ready. Could you please take a look at this PR? Thanks a lot.

I will take a look. Thanks.

File: docs/install/index.md [hunk 2]
```diff
@@ -32,6 +32,10 @@ pip install onnxruntime
 pip install onnxruntime-gpu
 ```
 
+```bash
+pip install onnxruntime-cann
```
Comment (jywu-msft) [inline docs/install/index.md:36]: these instructions are for installing onnxruntime packages available on pypi.
onnxruntime-cann isn't a published pypi package right? Please remove these lines.

**Comment by jywu-msft** [comment]
@faxu , @natke  can you help review for structure and consistency with rest of docs?

File: js/script.js [hunk 3]
```diff
@@ -1177,6 +1177,24 @@ var validCombos = {
 
     "linux,C++,X86,XNNPACK":
         "Follow build instructions from <a href='https://aka.ms/build-ort-xnnpack' target='_blank'>here</a>",
+
+    "linux,Python,ARM64,CANN":
+        "pip install onnxruntime-cann <br/>Refer to <a href='http://www.onnxruntime.ai/docs/execution-providers/CANN-ExecutionProvider.html#requirements' target='_blank'>docs</a> for requirements.",
```
Comment (jywu-msft) [inline js/script.js:1182]: are you guys planning on publishing an onnxruntime-cann package to pypi?

File: js/script.js [hunk 4]
```diff
@@ -1177,6 +1177,24 @@ var validCombos = {
...
+
+    "linux,C-API,ARM64,CANN":
+        "compile and install from source code <br/>Refer to <a href='http://www.onnxruntime.ai/docs/execution-providers/CANN-ExecutionProvider.html#build' target='_blank'>docs</a> for details.",
```
Comment (faxu) [inline js/script.js:1185]: For consistency, change these to "Follow build instructions from here<link>"

File: docs/install/index.md (hunk 2 above)
Comment (FFFrog) [reply docs/install/index.md:36]: We considered to publish onnxruntime-cann based on master branch of onnxruntime on pypi before. However, we now decided to publish it with version 1.13.0 of onnxruntime.

These lines have been removed and will be updated when 1.13.0 version are released

File: js/script.js (hunk 3 above)
Comment (FFFrog) [reply js/script.js:1182]: Yes, plan to publish it with version 1.13.0 of onnxruntime.

File: docs/build/eps.md [hunk 5]
```diff
@@ -17,7 +17,7 @@ redirect_from: /docs/how-to/build/eps
 
 ## Execution Provider Shared Libraries
 
-The oneDNN, TensorRT, and OpenVINO™ providers are built as shared libraries vs being statically linked into the main onnxruntime. This enables them to be loaded only when needed, and if the dependent libraries of the provider are not installed onnxruntime will still run fine, it just will not be able to use that provider. For non shared library providers, all dependencies of the provider must exist to load onnxruntime.
+The oneDNN, TensorRT, OpenVINO™ and CANN providers are built as shared libraries vs being statically linked into the main onnxruntime. This enables them to be loaded only when needed, and if the dependent libraries of the provider are not installed onnxruntime will still run fine, it just will not be able to use that provider. For non shared library providers, all dependencies of the provider must exist to load onnxruntime.
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply docs/build/eps.md]: Done, thanks a lot.

File: docs/execution-providers/CANN-ExecutionProvider.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
+---
+title: CANN (Huawei)
+description: Instructions to execute ONNX Runtime with the Huawei CANN execution provider
+parent: Execution Providers
+nav_order: 16
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply docs/execution-providers/CANN-ExecutionProvider.md]: Done, thanks a lot.

File: docs/execution-providers/CANN-ExecutionProvider.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+* TOC placeholder
+{:toc}
+
+## Requirements
+
+Please reference table below for official CANN packages dependencies for the ONNX Runtime inferencing package.
+
+|ONNX Runtime|CANN|OS|
+|---|---|---|---|
+|1.12|6.0|Ubuntu 18.04<br/>Ubuntu 20.04<br/>CentOS 7.8|
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply docs/execution-providers/CANN-ExecutionProvider.md:30]: yes, we have tested it and added a new record to the table above

File: docs/execution-providers/CANN-ExecutionProvider.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+|ai.onnx:Identity||
+|ai.onnx:MatMul|Input B should be constant.|
+|ai.onnx:MaxPool|Only 2D Pool is supported.|
+|ai.onnx:Mul||
+|ai.onnx:Relu||
+|ai.onnx:Sub||
+
+## Additional Resources
+
+We're keeping working on CANN EP enhancement. More feature, such as operatior support, performance tuning, will coming soon.
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply docs/execution-providers/CANN-ExecutionProvider.md]: Done, thanks a lot.

File: docs/execution-providers/index.md [hunk 9]
```diff
@@ -30,7 +30,7 @@ ONNX Runtime supports many different execution providers today. Some of the EPs
 ---|---|---|---
 |Default CPU|[NVIDIA CUDA](../execution-providers/CUDA-ExecutionProvider.md)|[Intel OpenVINO](../execution-providers/OpenVINO-ExecutionProvider.md)|[Rockchip NPU](../execution-providers/RKNPU-ExecutionProvider.md) (*preview*)|
 |[Intel DNNL](../execution-providers/oneDNN-ExecutionProvider.md)|[NVIDIA TensorRT](../execution-providers/TensorRT-ExecutionProvider.md)|[ARM Compute Library](../execution-providers/ACL-ExecutionProvider.md) (*preview*)|[Xilinx Vitis-AI](../execution-providers/Vitis-AI-ExecutionProvider.md) (*preview*)|
-|[TVM](../execution-providers/TVM-ExecutionProvider.md) (*preview*)|[DirectML](../execution-providers/DirectML-ExecutionProvider.md)|[Android Neural Networks API](../execution-providers/NNAPI-ExecutionProvider.md)||
+|[TVM](../execution-providers/TVM-ExecutionProvider.md) (*preview*)|[DirectML](../execution-providers/DirectML-ExecutionProvider.md)|[Android Neural Networks API](../execution-providers/NNAPI-ExecutionProvider.md)|[Huawei NPU](../execution-providers/CANN-ExecutionProvider.md) (*preview*)|
```
Comment (faxu) [inline docs/execution-providers/index.md:33]: For consistency, do you want to use "Huawei NPU" everywhere, or "CANN"?

**Comment by faxu** [comment]
There are a bunch of link errors in the CheckLinks test. Can you make sure your base is in sync with the latest in the gh-pages branch?

File: js/script.js [hunk 10]
```diff
@@ -1177,6 +1177,24 @@ var validCombos = {
 
     "linux,C++,X86,XNNPACK":
         "Follow build instructions from <a href='https://aka.ms/build-ort-xnnpack' target='_blank'>here</a>",
+
+    "linux,Python,ARM64,CANN":
+        "Follow build instructions from here <br/>Refer to <a href='http://www.onnxruntime.ai/docs/execution-providers/CANN-ExecutionProvider.html#build' target='_blank'>docs</a> for details.",
```
[the remark this answers is not in the document]
Comment (FFFrog) [reply js/script.js]: Have done.

File: docs/execution-providers/index.md (hunk 9 above)
Comment (FFFrog) [reply docs/execution-providers/index.md]: Thanks a lot.
“Huawei CANN” might be better.

**Comment by FFFrog** [comment]
> 



> There are a bunch of link errors in the CheckLinks test. Can you make sure your base is in sync with the latest in the gh-pages branch?

Yes，always in sync with gh-pages

File: js/script.js (hunk 3 above)
Comment (jywu-msft) [reply js/script.js]: please note there is a rel-1.3.1 branch. track that one instead of 1.3.0
thanks

File: index.html [hunk 11]
```diff
@@ -279,6 +279,8 @@ <h3 id="selectHardwareAcceleration">Hardware Acceleration</h3>
 															<span>Vitis AI (Preview)</span></div>
 														<div class="col-lg-3 col-md-3 r-option version" role="option" tabindex="-1" aria-selected="false" id="XNNPACK">
 															<span>XNNPACK (Preview)</span></div>
+														<div class="col-lg-3 col-md-3 r-option version" role="option" tabindex="-1" aria-selected="false" id="CANN">
+															<span>CANN (Preview)</span></div>
```
Comment (faxu) [inline index.html:283]: I thought you had moved this to alphabetical (should be after ArmNN (Preview)), but it reverted back to the end. Can you fix again?

Comment (FFFrog) [reply index.html]: sorry, it's my mistake.
The new commit to fix it is ready, please take a look at it again, thanks a lot.

**Comment by FFFrog** [comment]
sorry to @faxu @jywu-msft again, is there anything else that needs to be reviewed?
If so, I will fix it as soon as possible

**Review by faxu — APPROVED** [review]

**Comment by faxu** [comment]
LGTM. @jywu-msft and @natke - any additional comments before merging?

**Comment by natke** [comment]
LGTM :shipit:

**Comment by faxu** [comment]
> > LGTM. @jywu-msft and @natke - any additional comments before merging?
> 
> one thing i wasn't clear on. why did so many files need to be modified? looks like most of the changes to EP files are to the nav_order i think we can preserve existing nav_order and just increment for CANN

It's because we keep the EP list alphabetical for easier viewing.

**Pull request merged**